src/testcases/org/apache/poi/xssf/usermodel/helpers/XSSFColumnShifterTest.java

```diff
@@ -0,0 +1,384 @@
+        row.createCell(2, CellType.FORMULA).setCellFormula("B1-B3");
+        row = sheet1.createRow(4);
+        row.createCell(0, CellType.FORMULA).setCellFormula("SUM(C1:C4)");
+        row.createCell(1, CellType.FORMULA).setCellFormula("SUM(A3:C3)");
+        row.createCell(2, CellType.FORMULA).setCellFormula("$C1+C$2");
+        row = sheet1.createRow(5);
+        row.createCell(1, CellType.NUMERIC).setCellValue(1.5);
+        /*
+         * sheet2 = (XSSFSheet)wb07.createSheet("sheet2"); row =
+         * sheet2.createRow(0); row.createCell(0,
+         * CellType.NUMERIC).setCellValue(10); row.createCell(1,
+         * CellType.NUMERIC).setCellValue(11); row.createCell(2,
+         * CellType.FORMULA).setCellFormula("SUM(Sheet1!B3:C3)"); row =
+         * sheet2.createRow(1); row.createCell(0,
+         * CellType.NUMERIC).setCellValue(21); row.createCell(1,
+         * CellType.NUMERIC).setCellValue(22); row.createCell(2,
+         * CellType.NUMERIC).setCellValue(23); row = sheet2.createRow(2);
+         * row.createCell(0,
+         * CellType.FORMULA).setCellFormula("Sheet1!A4+Sheet1!C2+A2");
+         * row.createCell(1,
+         * CellType.FORMULA).setCellFormula("SUM(Sheet1!A3:$C3)"); row =
+         * sheet2.createRow(3); row.createCell(0,
+         * CellType.STRING).setCellValue("dummy");
+         */
+        // writeSheetToLog(sheet1);
+    }
+
+    @Test
+    public void testInsertOneColumn() {
+        sheet1.shiftColumns(1, 2, 1);
+        writeSheetToLog(sheet1);
+        String formulaA4 = sheet1.getRow(3).getCell(0).getCellFormula();
+        assertEquals("A2*C3", formulaA4);
+        String formulaC4 = sheet1.getRow(3).getCell(3).getCellFormula();
+        assertEquals("C1-C3", formulaC4);
+        String formulaB5 = sheet1.getRow(4).getCell(2).getCellFormula();
+        assertEquals("SUM(A3:D3)", formulaB5);
+        String formulaD5 = sheet1.getRow(4).getCell(3).getCellFormula(); // $C1+C$2
+        assertEquals("$D1+D$2", formulaD5);
+
+        String newb5Empty = sheet1.getRow(4).getCell(1).getStringCellValue();
+        assertEquals(newb5Empty, "");
+    }
+
+    @Test
+    public void testInsertTwoColumns() {
+        sheet1.shiftColumns(1, 2, 2);
+        String formulaA4 = sheet1.getRow(3).getCell(0).getCellFormula();
+        assertEquals("A2*D3", formulaA4);
+        String formulaD4 = sheet1.getRow(3).getCell(4).getCellFormula();
+        assertEquals("D1-D3", formulaD4);
+        String formulaD5 = sheet1.getRow(4).getCell(3).getCellFormula();
+        assertEquals("SUM(A3:E3)", formulaD5);
+
+        String b5Empty = sheet1.getRow(4).getCell(1).getStringCellValue();
+        assertEquals(b5Empty, "");
+        Object c6Null = sheet1.getRow(5).getCell(2); // null cell A5 is shifted
+                                                        // for 2 columns, so now
+                                                        // c5 should be null
+        assertEquals(c6Null, null);
+    }
```

Comment:
A few more test cases:

- shift left
- shift cells left of the first column (should throw some kind of exception, such as IllegalStateException or IllegalArgumentException)
- shift cells right of the last column
- Formulas appear to be well tested. How about numbers, strings, dates, cell styles?